test/e2e/upgrade_test.go

```diff
@@ -47,6 +47,22 @@ func TestVSphereKubernetes120UbuntuTo121Upgrade(t *testing.T) {
 	)
 }
 
+func TestVSphereKubernetes121UbuntuTo122Upgrade(t *testing.T) {
+	provider := framework.NewVSphere(t, framework.WithUbuntu121())
+	test := framework.NewClusterE2ETest(
+		t,
+		provider,
+		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube121)),
+	)
+	runSimpleUpgradeFlow(
+		test,
+		v1alpha1.Kube122,
+		framework.WithClusterUpgrade(api.WithKubernetesVersion(v1alpha1.Kube122)),
+		framework.WithEnvVar("K8S_1_22_SUPPORT", "true"),
```

Comment:
We have some stuff already in the e2e setup binary that set env vars depending on the test name. Or at least we used to for IAM tests

I would vote to do the same for these tests (if test name has "122" then set flag env var)

The main advantages are there will be only one place to cleanup and we keep all the feature flags for tests in one place